### DISCUSSION TITLE
Internationalized routing but different urls slugs (i18n)

### DISCUSSION THREAD

**Discussion by kimgiutae** [vercel/next.js#31178] [Help]
In the way that next.js docs approach i18n, **there is no way to translate url slugs**. (https://nextjs.org/docs/advanced-features/i18n-routing)

I mean... in an english/spanish website, 'pages/about.tsx' will convert to 'site.com/about' and 'site.com/es/about'. And the correct url should be 'site.com/es/nosotros' with the url translated too. Notice that "about" turns into "nosotros".

I found some workarounds that the community found, but seems not practical in mid/big multilingual sites. I mean, I can't write a redirect each time I wrote a post in my cms.

**I found easier for me as a developer to use the folder routing system of next.js to provide urls translations**.

I mean. To have a estructure like this:

pages/index.tsx (home english)
pages/about.tsx (about us english)
pages/news/[slug].tsx (news blog english)
pages/es/index.tsx (home spanish)
pages/es/nosotros.tsx (about us spanish)
pages/es/noticias/[slug].tsx (news blog spanish)

**What are the advantages/disadvantages of this approach?**

Right now I have no way to modify the lang attribute of document on server side depending of the current path.

Thanks in advance.

**Comment by kimgiutae** [comment]
Related issue https://github.com/vercel/next.js/discussions/18485

**Reply by wiiiimm** [reply]
Check out my reply at https://github.com/vercel/next.js/discussions/18485#discussioncomment-1719719.

Let's carry on our discussion over at #18485.
